core/data/pack/reader.go

```diff
@@ -42,13 +43,12 @@ func Read(ctx context.Context, from io.Reader, events Events, forceDynamic bool)
 		events: events,
 	}
 	r.pb = proto.NewBuffer(r.buf)
-	if err := r.readMagic(); err != nil {
+	if major, _, err := r.readHeader(); err != nil {
 		return err
+	} else if !(MinVersion <= major && major <= MaxVersion) {
+		return ErrUnsupportedVersion{Version: major}
```

Comment:
Why did you kill `Version` to then just ignore minor versions?

Reply:
Major version breaks file format, so it needs to be checked.
By definition, minion version, can not break format so shouldn't be checked.
Increment minor version can still be used in the future to indicate backward compatible changes.

Reply:
The error message displayed to the user includes the minor. You're dropping the minor version read from the file, so you're now potentially displaying an incorrect error message.
As you've just removed code that was perfectly fine, with something that could now display incorrect / confusing error messages, please put it back.

Reply:
The error message does not include the minor version.
Furthermore, error message displayed to the user is just limited to "too new" / "too old".

Reply:
Humm, perhaps I removed it due to the numbers not matching the releases.
I'd like the version mismatches to be printed to the logs at the very least, so please just keep the old version error type.